src/simsopt/field/mgrid.py

```diff
@@ -0,0 +1,290 @@
+import sys
+import numpy as np
+from scipy.io import netcdf_file
+
+__all__ = ["MGrid"]
+
+
+def _pad_string(string): 
+    '''
+    Pads a string with 30 underscores (for writing coil group names).
+    '''
+    return '{:^30}'.format(string).replace(' ', '_')
+
+
+def _unpack(binary_array):
+    '''
+    Decrypt binary char array into a string.
+    This function is used for reading coil group names.
+    '''
+    return "".join(np.char.decode(binary_array)).strip()
+
+
```

Comment:
This class is a good candidate for the descriptor protocol. I'll push the changes